ui/shared/components/panel/LoadWorkspaceDataForm.jsx

```diff
@@ -144,6 +171,11 @@ const LoadWorkspaceDataForm = React.memo(({ params }) => (
 
 LoadWorkspaceDataForm.propTypes = {
   params: PropTypes.object.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+}
+
+const mapDispatchToProps = {
```

Comment:
I don't like that we are connecting this component for both usages when it only needs to be connected for one of them. I think it would be better to leave this as a plain component and connect the component on the project page

Reply:
How about creating a connected component dedicated for loading a new dataset? It is a little tricky to connect the component on the project page while using a reducer function (addDataFromWorkspace) from this file.